adapters/repos/db/lsmkv/bucket.go

```diff
@@ -324,7 +324,8 @@ func (b *Bucket) IterateObjects(ctx context.Context, f func(object *storobj.Obje
 	i := 0
 
 	for k, v := cursor.First(); k != nil; k, v = cursor.Next() {
-		obj, err := storobj.FromBinary(v)
+		// TODO: investigate the callers of this method
+		obj, err := storobj.FromBinary_exp(v)
```

Comment:
Most of the callsites require full objects, however, one place could be optimized to unmarshal only vectors:
```
func (m *Migrator) RecalculateVectorDimensions(ctx context.Context) error {
```